fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/domain/JobExecutionRepository.java

```diff
@@ -165,30 +170,85 @@ public void updateJobStatusToFailed(Long stuckJobId, String partitionerStepName)
-            return "SELECT cjp.id FROM (SELECT id,json_array_elements(parameter_json) AS json_data FROM batch_custom_job_parameters) AS cjp WHERE (cjp.json_data ::jsonb @> :jsonString ::jsonb)";
-        } else {
-            throw new IllegalStateException("Database type is not supported for json query " + databaseTypeResolver.databaseType());
+        String businessDateInString;
+        try {
+            if (databaseTypeResolver.isPostgreSQL()) {
+                businessDateInString = namedParameterJdbcTemplate.queryForObject("""
+                        SELECT "parameterValue" FROM json_populate_recordset(null::record,
+                        (SELECT
+                            CJP.parameter_json
+                        FROM
+                            BATCH_JOB_INSTANCE BJI
+                                INNER JOIN BATCH_JOB_EXECUTION BJE ON BJI.JOB_INSTANCE_ID = BJE.JOB_INSTANCE_ID
+                                INNER JOIN BATCH_JOB_EXECUTION_PARAMS BJEP ON BJE.JOB_EXECUTION_ID = BJEP.JOB_EXECUTION_ID
+                                inner join batch_custom_job_parameters CJP ON cast(BJEP.parameter_value as bigint) = CJP.id
+                                AND BJEP.parameter_name = :jobCustomParamKeyName
+                                CROSS JOIN LATERAL json_array_elements(CJP.parameter_json) J
+                        WHERE
+                                    J ->> 'parameterName' = :parameterKeyName
+                          AND J ->> 'parameterValue' = :parameterValue
+                          AND BJE.STATUS IN (:statuses)
+                          AND BJI.JOB_NAME = :jobName
+                          AND BJE.JOB_INSTANCE_ID NOT IN (
+                            SELECT
+                                IBJE.JOB_INSTANCE_ID
+                            FROM
+                                BATCH_JOB_INSTANCE IBJI
+                                    INNER JOIN BATCH_JOB_EXECUTION IBJE ON IBJI.JOB_INSTANCE_ID = IBJE.JOB_INSTANCE_ID
+                            WHERE
+                                    IBJE.STATUS IN (:completedStatuses)
+                              AND IBJI.JOB_NAME = :jobName
+                        )) AS ("parameterName" varchar(100), "parameterValue" varchar(100)) where "parameterName" = 'BusinessDate';
+                        """,
+                        Map.of("jobCustomParamKeyName", jobCustomParamKeyName, "parameterKeyName", parameterKeyName, "parameterValue",
+                                parameterValue, "statuses", List.of(STARTED.name(), STARTING.name()), "completedStatuses",
+                                List.of(COMPLETED.name(), FAILED.name(), UNKNOWN.name()), "jobName", jobName),
+                        String.class);
+            } else if (databaseTypeResolver.isMySQL()) {
+                businessDateInString = namedParameterJdbcTemplate.queryForObject(
+                        """
+                                SELECT PARAMS.parameter_value FROM json_table(
```

Comment:
Is this works if multiple paramKey-paramValue entries can be found in the json?

Reply:
you mean multiple paramameters with the "BusinessDate" parameter name?

Reply:
no. i mean multiple entry in the json like
[
{
"parameterKey": "randomKey",
"parameterValue": "randomValue"
},
{
"parameterKey": "BusinessDate",
"parameterValue": "2023-07-08"
}
]

i reckon we can have multiple param key-value pair in the json, no?

Reply:
ther is a WHERE clause at the very end of the query that filters for only the 'BusinessDate' parameter